pkg/kube/configmap/configmap.go

```diff
@@ -87,3 +90,38 @@ func CreateOrUpdate(getUpdateCreator GetUpdateCreator, cm corev1.ConfigMap) erro
 	}
 	return getUpdateCreator.UpdateConfigMap(cm)
 }
+
+// filelikePropertiesToMap converts a file-like field in a ConfigMap to a map[string]string.
+func filelikePropertiesToMap(s string) (map[string]string, error) {
+	keyValPairs := map[string]string{}
+	s = strings.TrimRight(s, "\n")
+	for _, keyValPair := range strings.Split(s, "\n") {
+		splittedPair := strings.Split(keyValPair, "=")
```

Comment:
[nit] maybe we could have package private constants for `=` and `\n` to help readability, what do you think?